engine/core/src/main/java/org/datacleaner/job/CompoundComponentRequirement.java

```diff
@@ -144,11 +185,8 @@ public boolean equals(Object obj) {
         if (getClass() != obj.getClass())
             return false;
         final CompoundComponentRequirement other = (CompoundComponentRequirement) obj;
-        if (_outcomes == null) {
-            if (other._outcomes != null)
-                return false;
-        } else if (!_outcomes.equals(other._outcomes))
-            return false;
-        return true;
+
+        // TODO: This wasn't anything equivalent to deepEquals before... But should it have been?
```

Comment:
Ah, yeah. True in most cases (including this). AbstractCollection makes no such guarantee, and it's not in the Collection contract, but AbstractSet and AbstractList does.